python/cuml/dask/common/base.py

```diff
@@ -0,0 +1,192 @@
+
+        parallelism : int
+            Amount of concurrent partitions that will be processed
+            per worker. This bounds the total amount of temporary
+            workspace memory on the GPU that will need to be allocated
+            at any time.
+
+        output_futures : bool returns the futures pointing the to the resuls
+                         of the parallel function executions on the workers,
+                         rather than a dask collection object.
+
+        Returns
+        -------
+        y : dask cuDF (n_rows, 1)
+        """
+
+        if delayed:
+            X_d = X.to_delayed()
+
+            lock = dask.delayed(MultiHolderLock(parallelism),
+                                pure=True)
+
+            model = dask.delayed(self.local_model, pure=True)
+
+            func = dask.delayed(func, pure=False, nout=1)
+
+            if isinstance(X, dcDataFrame):
+                preds = [func(model, lock, part) for part in X_d]
+                return preds if output_futures \
+                    else dask.dataframe.from_delayed(preds)
+
+            else:
+                preds = [func(model, lock, part[0])
+                         for part in X_d]
+
+                # todo: add parameter for option of not checking directly
+                dtype = X.dtype
+
+                preds_arr = [
+                    dask.array.from_delayed(pred,
+                                            meta=cp.zeros(1, dtype=dtype),
+                                            shape=(np.nan,),
+                                            dtype=dtype)
+                    for pred in preds]
+
+                if output_futures:
+                    return preds_arr
+                else:
+                    return dask.array.concatenate(preds_arr, axis=0,
+                                                  allow_unknown_chunksizes=True
+                                                  )
+
+        else:
+            X = X.persist()
+
+            data = MGData.single(X, client=self.client)
+            scattered = self.client.scatter(self.local_model,
+                                            workers=data.workers,
+                                            broadcast=True,
+                                            hash=False)
+
+            lock = self.client.scatter(MultiHolderLock(parallelism),
+                                       workers=data.workers,
+                                       broadcast=True,
+                                       hash=False)
+
+            key = uuid1()
+            func_futures = [self.client.submit(
+                func,
+                scattered,
+                lock,
+                p,
+                key=key) for w, p in data.gpu_futures]
+
+            return func_futures if output_futures \
+                else to_output(func_futures, self.datatype)
+
+
+class DelayedPredictionMixin(DelayedParallelFunc):
+
+    def _predict(self, X, delayed=True, parallelism=25):
```

Comment:
just a very small thing but i believe there is a typo in the default setting for prarllelism either here or in the dask code of the models.

Reply:
Also if it was possible to pass in more variables to the predict function that would be awesome since RF predict requires more parameters.

Reply:
@Salonijain27 just did a quick push to enable this for you, let me know if you need any other changes

Reply:
Thank you!